django/contrib/auth/forms.py

```diff
@@ -94,7 +94,8 @@ class Meta:
 
     def __init__(self, *args, **kwargs):
         super(UserCreationForm, self).__init__(*args, **kwargs)
-        self.fields[self._meta.model.USERNAME_FIELD].widget.attrs.update({'autofocus': ''})
+        if hasattr(self, 'fields') and self._meta.model.USERNAME_FIELD in self.fields:
```

Comment:
I think the `hasattr` is unnecessary as the check is after `super().__init__()`. A child class of `BaseForm` will always have a `fields` instance variable, right? Removing the `hasattr` does not cause any tests to fail.